src/routes/wohnen/components/Offer.js

```diff
@@ -0,0 +1,137 @@
+// @flow
+
+import React from 'react'
+import WohnenOfferModel from 'modules/endpoint/models/WohnenOfferModel'
+import WohnenFormData from 'modules/endpoint/models/WohnenFormData'
+import styled from 'styled-components'
+import type { AccommodationType } from '../../../modules/endpoint/models/WohnenFormData'
+import type { TFunction } from 'react-i18next'
+import { translate } from 'react-i18next'
+import ListElement from '../../../modules/common/components/ListElement'
+
+type PropsType = {
+  offer: WohnenOfferModel<*>,
+  t: TFunction
+}
+
+const Header = styled.div`
+  padding: 15px 5px 5px;
+  font-weight: 700;
+`
+
+const RowTitle = styled.div`
+  flex: 50%;
+`
+
+const RowValue = styled.div`
+  flex: 50%;
+`
+
+const Row = styled.div`
+  display: flex;
+`
+
+class Offer extends React.Component<PropsType> {
+  translate (type: 'runningServices' | 'additionalServices' | 'rooms', keys: Array<string>): Array<string> {
+    return keys.map(key => this.props.t(`values.${type}.${key}`))
+  }
+
+  stringify (words: Array<string>): string {
+    return words.join(', ')
+  }
+
+  render () {
+    const offer = this.props.offer
+
+    if (offer.formDataType === WohnenFormData) {
+      const specificOffer: WohnenOfferModel<WohnenFormData> = offer
+      const accommodation: AccommodationType = specificOffer.formData.accommodation
+      const costs = specificOffer.formData.costs
+      const landlord = offer.formData.landlord
+
+      const translateRunningServices = keys => this.stringify(this.translate('runningServices', keys))
+      const translateAdditionalServices = keys => this.stringify(this.translate('additionalServices', keys))
+      const translateRooms = keys => this.stringify(this.translate('rooms', keys))
+
+      return <React.Fragment>
+        <div>
```

Comment:
If you want you could split this in different components, which would make it way more clear.

Reply:
My point was not that it would be reusable but that it is quite a lot of stuff in one file, but its okay.